waku/v2/node/wakunode2.nim

```diff
@@ -125,6 +125,22 @@ proc removeContentFilters(filters: var Filters, contentFilters: seq[ContentFilte
   
   debug "filters modified", filters=filters
 
+proc updateSwitchPeerInfo(node: WakuNode) =
+  ## TODO: remove this when supported upstream
+  ## 
+  ## nim-libp2p does not yet support announcing addrs
+  ## different from bound addrs.
+  ## 
+  ## This is a temporary workaround to replace
+  ## peer info addrs in switch to announced
+  ## addresses.
+  ## 
+  ## WARNING: this should only be called once the switch
```

Comment:
In peerInfo.addrs an empty addrs would be indication that the switch is not built before and can be used to prevent update if switch is not built.

Reply:
Right, I don't think it will indicate though if the switch has been `start`ed or not? Which is what we want to avoid, as the switch needs to start its transports on the bind addresses, not external.